internal/clients/http/ec2/ec2_client.go

```diff
@@ -442,6 +485,52 @@ func getRoleName(arn string) (string, error) {
 	return roleName[1], nil
 }
 
+func getStatementFromAWSDocument(document string) ([]Statement, error) {
+	unescapedDoc, err := url.QueryUnescape(document)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse url from the Document: %w", err)
+	}
+
+	var policyVersion Version
+	err = json.Unmarshal([]byte(unescapedDoc), &policyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("Could not unmarshal json: %w", err)
+	}
+	return policyVersion.Statement, nil
+}
+
+func listStatements(versions []*iamTypes.PolicyVersion) ([]Statement, error) {
+	var result []Statement
+	for _, version := range versions {
+		policyVersionStatements, err := getStatementFromAWSDocument(*version.Document)
+		if err != nil {
+			return nil, fmt.Errorf("Could not fetch statement from AWS document: %w", err)
+		}
+
+		result = append(result, policyVersionStatements...)
+	}
+	return result, nil
+}
+
+func listMissingPermissions(statements []Statement, expected Statement) []string {
+	presentPermissions := make(map[string]struct{})
+	var missing []string
```

Comment:
ditto

Reply:
Same, we do not know how many permissions would be missing. Therefore a slice seemed more convenient. This should work the same if the nil is returned instead of an empty array.
Let me know if this makes sense. I am happy to reconsider.